content/en/docs/concepts/overview/kubernetes-api.md

```diff
@@ -102,13 +102,21 @@ to ensure that the API presents a clear, consistent view of system resources
 and behavior, and to enable controlling access to end-of-life and/or
 experimental APIs.
 
-Refer to [API versions reference](/docs/reference/using-api/api-overview/#api-versioning)
-for more details on the API version level definitions.
-
 To make it easier to evolve and to extend its API, Kubernetes implements
 [API groups](/docs/reference/using-api/api-overview/#api-groups) that can be
 [enabled or disabled](/docs/reference/using-api/api-overview/#enabling-or-disabling).
 
+API resources are distinguished from each other based on their API groups and
+their resource types, which is referred to as their `kind`s. The API server may
+serve more than one API version for the same resource. All these different
+versions are simply representations of the same resource. Suppose there are two
+versions `v1` and `v1beta1` for the same resource, an object created via the
+`v1beta1` version can then be read, updated, and deleted via the `v1` version.
+The API server handles the conversion between API versions transparently.
```

Comment:
Could you use singular instead of plural:
An API resource is distinguished from other resources by their API group and
their API resource type which is referred to as their `kind`.


I need to read the version doc in more detail.
Does the conversion rule hold for all resources? Is the automatic conversion only from a v1beta version to
a v1 version?

Reply:
According to @liggitt  (kubernetes/kubernetes#94749), different versions of a resource are just different representations of the same resource. The conversion works in both directions. I wasn't aware of this before so I thought it would be good to call this out in the doc.

Reply:
OK. Seems like there could be hiccups in the conversion (missing fields, default values). I'll look at that PR.

Reply:
The SIG APIMachinery has spent a lot of time on ensuring this conversion. There are defaulting rules enforced, located at `pkg/apis/<group>/<version>/defaults.go`; there are conversion logic between a specific API version and the stored version, located at `pkg/apis/<group>/<version>/conversion.go`. For all changes to resource definitions, the team has been striving to ensure that object written via one version can be read from another version.

If you are interested, you may want to check one of the PRs that adds a 'signerName' to the 'CertificateSigningRequest'. 
(https://github.com/kubernetes/kubernetes/commit/a983356caa9876079eafc5e8dd7806473022e958)

Reply:
Thanks for the information!